Brokerages/Oanda/OandaBrokerage.cs

```diff
@@ -0,0 +1,1052 @@
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OandaBrokerage"/> class.
+        /// </summary>
+        /// <param name="orderProvider">The order provider.</param>
+        /// <param name="holdingsProvider">The holdings provider.</param>
+        /// <param name="accountId">The account identifier.</param>
+        public OandaBrokerage(IOrderProvider orderProvider, IHoldingsProvider holdingsProvider, int accountId)
+            : base("Oanda Brokerage")
+        {
+            _orderProvider = orderProvider;
+            _holdingsProvider = holdingsProvider;
+            AccountId = accountId;
+            _cachedOpenOrdersByOandaOrderId = new ConcurrentDictionary<long, DataType.Order>();
+            InstrumentSecurityTypeMap =  new Dictionary<string, SecurityType>(); 
+        }
+
+        /// <summary>
+        /// Initializes the instrument security type map.
+        /// </summary>
+        public void InitializeInstrumentSecurityTypeMap()
+        {
+            InstrumentSecurityTypeMap = MapInstrumentToSecurityType(GetInstrumentsAsync());
+        }
+
+        /// <summary>
+        /// Returns true if we're currently connected to the broker
+        /// </summary>
+        public override bool IsConnected
+        {
+            get { return _issuedAt + _lifeSpan > DateTime.Now; }
+        }
+
+        /// <summary>
+        /// Gets all open orders on the account. 
+        /// NOTE: The order objects returned do not have QC order IDs.
+        /// </summary>
+        /// <returns>The open orders returned from Oanda</returns>
+        public override List<Order> GetOpenOrders()
+        {
+            var oandaOrders = GetOrderList();
+
+            var orderList = oandaOrders.Select(ConvertOrder).ToList();
+            foreach (var openOrder in orderList)
+            {
+                _cachedOpenOrdersByOandaOrderId[openOrder.Id] = oandaOrders.FirstOrDefault(oo => oo.id == openOrder.Id);
+            }
+            return orderList;
+        }
+
+        /// <summary>
+        /// Gets all holdings for the account
+        /// </summary>
+        /// <returns>The current holdings from the account</returns>
+        public override List<Holding> GetAccountHoldings()
+        {
+            var holdings = GetPositions().Select(ConvertHolding).ToList();
+            return holdings;
+        }
+
+        /// <summary>
+        /// Converts the Oanda position into a QuantConnect holding.
+        /// </summary>
+        /// <param name="position">The position.</param>
+        /// <returns></returns>
+        protected Holding ConvertHolding(Position position)
+        {
+            return new Holding
+            {
+                Symbol = position.instrument,
+                Type = InstrumentSecurityTypeMap[position.instrument],
+                AveragePrice = (decimal)position.avgPrice,
+                ConversionRate = 1.0m,
+                CurrencySymbol = "$",
```

Comment:
We use the USD conversion for computing the portfolio value, but we do store currencies in their native units, for example, if an account holds 10k CHF, we'll store it as 10k CHF, but we stil need the CHF->USD conversion for portfolio value.A little more complicated, if the account has a forex position, say 10k EUR/GBP, that's really the client sold GBP for EUR. We would need the EUR->GBP rate (the market price of the position) as well as the GBP->USD rate so we could compute portfolio value.